File: cpp/LSP/step1/Car.cpp
#include <exception>
#include <string>
#include "Vehicle.cpp"

using namespace std;

class Car : public Vehicle {
public:
    void ChangeGear(Gear gear) override {
        if (Gear::R == gear && GetGear() == Gear::D) {
            string exception_msg = string("Can't change to REVERSE gear when ") + GetGearName(gear) + string(" gear is engaged!");
            throw exception(exception_msg.c_str());
        }
    }
};

File: python/LSP/step1/car.py
from vehicle import Vehicle, Gear

class Car(Vehicle):
    def change_gear(self, gear: Gear):
        if Gear.R == gear and self.get_gear() == Gear.D :
            exception_msg = 'Can\'t change to REVERSE gear when ' + self.get_gear_name(gear) + ' gear is engaged!'
            raise Exception(exception_msg)
        super().change_gear(gear)
